### PR TITLE
added detailed post endpoint. Only fetches top level comments. Unit t…

### PR DESCRIPTION
**Pull request by kaplunket**
…est added.

Part of #41

## Proposed Changes
  -Added endpoint for detailed post
  -Added a unit test
  -

### CODE REVIEW

**Comment by andrewh318** [comment]
Looking at the MongoDB docs, I still think we should follow a pattern similar to this and have a 'ref' to the 'Comment'. Doing this, we can automatically fetch all the comments for a post using the `.populate()` method. What are your thoughts? 
https://mongoosejs.com/docs/populate.html

File: server/src/routes/posts.js [hunk 1]
```diff
@@ -40,6 +40,28 @@ router.put('/:id', async (req, res) => {
...
+      title: foundPost.title,
+      body: foundPost.body,
+      upvotes_clap: foundPost.upvotes_clap,
+      upvotes_laugh: foundPost.upvotes_laugh,
+      upvotes_sad: foundPost.upvotes_sad,
+      date_created: foundPost.createdAt,
+      comments: await Comment.find({ _id: { $in: [foundPost.comment_id] } }),
+    };
+
+    res.send(postData);
```
Comment (GrunWong) [inline server/src/routes/posts.js]: Do we need a response status code here?

Comment (kaplunket) [reply server/src/routes/posts.js]: It returns 200 based on the unit tests

**Comment by kaplunket** [comment]
Do we want to leave changing to populate for a later group/issue?

File: server/src/routes/posts.js (hunk 1 above)
Comment (andrewh318) [reply server/src/routes/posts.js]: > Do we want to leave changing to populate for a later group/issue?

Yeah, I've removed my comment also. I realised if we want to support multi nested comments in the future as well, I"m not sure if the populate method will still work.

**Comment by andrewh318** [comment]
> Do we want to leave changing to populate for a later group/issue?

Yeah, I've removed my comment also. I realised if we want to support multi nested comments in the future as well, I"m not sure if the populate method will still work.

File: server/src/routes/posts.js (hunk 1 above)
Comment (GrunWong) [reply server/src/routes/posts.js]: Cool! Should be fine then.

**Comment by GrunWong** [comment]
@andrewh318 are you happy with the work Kazuki has done? Am I merge it into the master?

**Comment by andrewh318** [comment]
> @andrewh318 are you happy with the work Kazuki has done? Am I merge it into the master?

Yup, LGTM!

**Pull request merged**